### PR TITLE
Add encode methods to common structs

### PR DESCRIPTION
**Pull request by Rimpampa**
Add `encode` methods to the structs that contain information present in PNG chunks.

Originated from [PR#255](https://github.com/image-rs/image-png/pull/255).

### CODE REVIEW

File: src/common.rs [hunk 1]
```diff
@@ -343,6 +369,10 @@ impl ScaledFloat {
     pub fn into_value(self) -> f32 {
         Self::reverse(self.0) as f32
     }
+
+    pub fn encode<W: Write>(self, w: &mut W) -> encoder::Result<()> {
+        encoder::write_chunk(w, chunk::gAMA, &self.into_scaled().to_be_bytes())
+    }
```
Comment (HeroicKatora) [inline src/common.rs:375]: `ScaledFloat` is used in different chunks, exposing `encode` such that it encodes it as a gama chunk could be confusing.
```suggestion
    pub(crate) fn encode_gama<W: Write>(self, w: &mut W) -> encoder::Result<()> {
        encoder::write_chunk(w, chunk::gAMA, &self.into_scaled().to_be_bytes())
    }
```

**Review by HeroicKatora — CHANGES_REQUESTED** [review]
I just have one little nit.

File: src/encoder.rs [hunk 2]
```diff
@@ -253,67 +253,11 @@ impl<W: Write> Writer<W> {
...
-            if let Some(g) = &self.info.source_gamma {
-                write_chunk(&mut self.w, chunk::gAMA, &g.into_scaled().to_be_bytes())?;
-            }
-
-            if let Some(c) = &self.info.source_chromaticities {
-                let enc = Self::chromaticities_to_be_bytes(&c);
-                write_chunk(&mut self.w, chunk::cHRM, &enc)?;
-            }
-        }
+        self.info.encode(&mut self.w)?;
```
Comment (HeroicKatora) [inline src/encoder.rs:256]: Seems like you've accidentally dropped writing the `png` signature itself :joy:

Comment (Rimpampa) [reply src/encoder.rs:256]: Yeah just noticed😅

**Review by HeroicKatora — APPROVED** [review]

**Pull request merged**